use FileFilter in directory update

tisoft opened the pull request.

ricardojlrufino commented on src/main/java/com/ricardojlrufino/jexplorer/FileTreeModel.java:
```diff
@@ -18,10 +21,17 @@
 
 public class FileTreeModel extends DefaultTreeModel {
 
-  public FileTreeModel(TreeNode root) {
```
Please keep this constructor

tisoft commented:
I have rewritten the whole file filter handling and added a filter textbox to the FileExplorerPanelDemo

I don't have a public project right now, that is using it.
